providers/os/connection/device/linux/device_manager.go

```diff
@@ -91,15 +92,15 @@ func (d *LinuxDeviceManager) IdentifyMountTargets(opts map[string]string) ([]*sn
 		partitions = append(partitions, partitionsForDevice...)
 	}
 
-	if err := errors.Join(errs...); err != nil {
-		return partitions, err
-	}
-
 	if opts[SkipAttemptExpandPartitions] == "true" {
+		log.Warn().Msg("!!! skipping attempt to expand partitions")
 		return partitions, nil
 	}
 
-	return d.attemptExpandPartitions(partitions)
+	partitions, err = d.attemptExpandPartitions(partitions)
```

Comment:
since we're not returning on err anymore, does `attemptExpandPartitions` protect against possible nil arrays being passed in?